x-pack/plugins/security_solution/common/detection_engine/schemas/types/threat_mapping.ts

```diff
@@ -19,29 +19,41 @@ export type ThreatFilters = t.TypeOf<typeof threat_filters>;
 export const threatFiltersOrUndefined = t.union([threat_filters, t.undefined]);
 export type ThreatFiltersOrUndefined = t.TypeOf<typeof threatFiltersOrUndefined>;
 
-export const threatMappingEntries = t.array(
-  t.exact(
-    t.type({
-      field: NonEmptyString,
-      type: t.keyof({ mapping: null }),
-      value: NonEmptyString,
-    })
-  )
+export const threatMapEntry = t.exact(
+  t.type({
+    field: NonEmptyString,
+    type: t.keyof({ mapping: null }),
+    value: NonEmptyString,
+  })
 );
+
+export type ThreatMapEntry = t.TypeOf<typeof threatMapEntry>;
+
+export const threatMappingEntries = t.array(threatMapEntry);
 export type ThreatMappingEntries = t.TypeOf<typeof threatMappingEntries>;
 
-export const threat_mapping = t.array(
-  t.exact(
-    t.type({
-      entries: threatMappingEntries,
-    })
-  )
+export const threatMap = t.exact(
+  t.type({
+    entries: threatMappingEntries,
```

Comment:
Could this be made to be a non empty array? I just thought of that when I saw the `containsEmptyItem` helper. I remember having to do lots of checks like that with exceptions and finally just changed it so that it's required to not be empty since at the very least (in the UI) there's one item with empty values.

Reply:
Let me add that to a potential follow up. I totally agree, just don't know about the time I have left with the other tasks but I will add it to my list as I will have to make one of those icky specific types and all the tests for it.